### ISSUE TITLE
Method exclusion doesn't work

### ISSUE THREAD

**Issue by rmetzger** [siom79/japicmp#157]
In our Japicmp config, we have the following exclude:

```
	<configuration>
					<parameter>
						<excludes combine.children="append">
							<!-- Exclude generated classes from api compatibility checks -->
							<exclude>*\$\$anon\$*</exclude>
							<exclude>org.apache.flink.streaming.api.scala.WindowedStream#aggregate(org.apache.flink.streaming.api.functions.aggregation.AggregationFunction$AggregationType,int)</exclude>
							<exclude>org.apache.flink.streaming.api.scala.AllWindowedStream#aggregate(org.apache.flink.streaming.api.functions.aggregation.AggregationFunction$AggregationType,int)</exclude>
						</excludes>
					</parameter>
				</configuration>
```

As you can see from the report, the exclude is understood by the plugin, but still, the method (with the exact same signature) is marked as missing.
http://people.apache.org/~rmetzger/japicmp.html

This is the branch were the error is happening: https://github.com/rmetzger/flink/tree/agg_follow_ups

**Comment by siom79** [comment]
Thanks for reporting. I used regular expressions for exclude matching but did not escape the dollar sign in inner classes.

Resolved on `master` branch. Will be released with 0.9.4.

**Comment by rmetzger** [comment]
Thanks a lot for looking into the issue so quickly!

Looking forward to the 0.9.4 release.

**Comment by siom79** [comment]
Meanwhile you can escape the dollar sign in your exclude: '\$'.

**Comment by StephanEwen** [comment]
Thanks for the fast response!

**Comment by siom79** [comment]
Released with 0.9.4.
